### DISCUSSION TITLE
User Segmentation / Custom Attributes

### DISCUSSION THREAD

**Discussion by melchiorb** [umami-software/umami#2608] [Ideas]
Hello,

I'm trying to use Umami for an upcoming web app and I love it, but I'm running into one serious obstacle: the lack of user segmentation.

I would like to be able to segment logged in and anonymous users and I can't find a good way to do so.

For an app that has registered users it is kind of critical to be able to tell the difference between your users and anonymous people.

Is there a plan to support this in the future? Even one fixed field that could be populated with different values would be hugely helpful.

I was trying to append the authentication status as a query parameter to all urls via manual tracking but it's really hacky and the filtering in the UI is still very limited.

I know arbitrary fields can be tricky for DB performance but even a single, fixed field would make a dramatic difference for sites with logged-in users.

Thanks!

**Comment by mikecao** [comment]
We have an upcoming feature that should help you with this: session data. You'll be able to attach data to visitors so you can track whatever you want, including logged in status.

**Reply by melchiorb** [reply]
Thanks for the reponse, that sounds perfect.

Is it the "ume-660-implement-new-session-visitor" branch?

We'd be happy to beta test this feature.

**Reply by mathix420** [reply]
Would also be more than happy to beta test!

**Reply by mathix420** [reply]
could be great if you had a public roadmap too 😉
